File: frontend/src/microservices/traveler/objects/TravelerServiceSummary.ts
// 本文件定义 traveler 模块的 `TravelerServiceSummary`，作为服务摘要传输数据并提供 JSON 编解码。

export type TravelerServiceSummary = {
  age: number | null
  documentLabel: string
  contactLabel: string
  preferenceLabel: string
  requirementLabel: string
  warningLevel: string
}

export const travelerServiceSummaryFromJson = (json: string): TravelerServiceSummary =>
  JSON.parse(json) as TravelerServiceSummary

export const travelerServiceSummaryToJson = (value: TravelerServiceSummary): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/traveler/objects/TravelerServiceSummary.scala
// 本文件定义 traveler 模块的服务摘要对象 `TravelerServiceSummary`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerServiceSummary(
    age: Option[Int],
    documentLabel: String,
    contactLabel: String,
    preferenceLabel: String,
    requirementLabel: String,
    warningLevel: String
)

object TravelerServiceSummary:
  given sourceEncoder: Encoder[TravelerServiceSummary] = deriveEncoder
  given sourceDecoder: Decoder[TravelerServiceSummary] = deriveDecoder
